sesman/session.c

```diff
@@ -420,15 +349,51 @@ wait_for_xserver(int display)
 }
 
 /******************************************************************************/
+static int
+session_start_chansrv(char *username, int display)
+{
+    struct list *chansrv_params;
+    char exe_path[262];
+    int cspid;
+
+    cspid = g_fork();
+    if (cspid == 0)
+    {
+        chansrv_params = list_create(); 
+        chansrv_params->auto_free = 1;
+
+        /* building parameters */
+        g_snprintf(exe_path, 261, "%s/xrdp-chansrv", XRDP_SBIN_PATH);
```

Comment:
Please use sizeof(exe_path) instead of 261. g_snprintf() and underlying vsnprintf() count the final zero towards the maximum length.